oauth2_http/java/com/google/auth/oauth2/UserAuthorizer.java

```diff
@@ -470,6 +508,19 @@ public void onChanged(OAuth2Credentials credentials) throws IOException {
     }
   }
 
+  /**
+   * A helper function to get a base64 encoded basic auth.
+   *
+   * @param clientId A ClientId object which contains a string value and a secret
+   * @return The basic auth header
+   */
+  public static String getBasicAuthString(ClientId clientId) {
+    String encodedCredentials =
+        BaseEncoding.base64()
+            .encode((clientId.getClientId() + ":" + clientId.getClientSecret()).getBytes());
+    return "Basic " + encodedCredentials;
+  }
```

Comment:
nit: Can you make this either private scope or move it to a public util class (if there exists one)?

Reply:
Make sense. Moved to `OAuth2Utils`